src/tx-format/transaction.md

```diff
@@ -193,3 +194,40 @@ Transaction is invalid if:
+
+When the bytecode is completed it can be used to upgrade the network.
+
+The size of each subsection can be arbitrary; the only limit is the maximum number of subsections allowed by the network. The combination of the transaction gas limit and the number of subsections limits the final maximum size of the bytecode.
+
+| name                | type                        | description                                                                             |
+|---------------------|-----------------------------|-----------------------------------------------------------------------------------------|
+| `root`              | `byte[32]`                  | The root of the Merkle tree is created over the bytecode.                               |
+| `witnessIndex`      | `uint16`                    | The witness index of the subsection of the bytecode.                                    |
+| `subsectionIndex`   | `uint16`                    | The index of the subsection of the bytecode.                                            |
+| `subsectionsNumber` | `uint16`                    | The total number of subsections on which bytecode was divided.                          |
+| `proofSetCount`     | `uint16`                    | Number of Merkle nodes in the proof.                                                    |
+| `policyTypes`       | `uint32`                    | Bitfield of used policy types.                                                          |
+| `inputsCount`       | `uint16`                    | Number of inputs.                                                                       |
+| `outputsCount`      | `uint16`                    | Number of outputs.                                                                      |
+| `witnessesCount`    | `uint16`                    | Number of witnesses.                                                                    |
+| `proofSet`          | `byte[32][]`                | The proof set of Merkle nodes to verify the connection of the subsection to the `root`. |
+| `policies`          | [Policy](./policy.md)`[]`   | List of policies.                                                                       |
+| `inputs`            | [Input](./input.md)`[]`     | List of inputs.                                                                         |
+| `outputs`           | [Output](./output.md)`[]`   | List of outputs.                                                                        |
+| `witnesses`         | [Witness](./witness.md)`[]` | List of witnesses.                                                                      |
+
+Transaction is invalid if:
```

Comment:
If I understand correctly, we also need this:

> - `subsectionIndex >= subsectionsNumber`

Also, is there a way to verify that subsectionsNumber doesn't change between Upload txs?

Reply:
`subsectionIndex >= subsectionsNumber` is enforced by the root verification, but I added it just for safety=)
The verification of the proof should guarantee the same `subsectionsNumber`. Changing `subsectionsNumber` affects the Merkle tree path.

Right now we have a bug in the `verify` implementation: https://github.com/FuelLabs/fuel-vm/issues/716